src/storage/v3/vertex.hpp

```diff
@@ -13,24 +13,28 @@
 
 #include <limits>
 #include <tuple>
+#include <type_traits>
 #include <vector>
 
 #include "storage/v3/delta.hpp"
 #include "storage/v3/edge_ref.hpp"
 #include "storage/v3/id_types.hpp"
+#include "storage/v3/key_store.hpp"
 #include "storage/v3/property_store.hpp"
+#include "storage/v3/property_value.hpp"
 #include "utils/spin_lock.hpp"
 
 namespace memgraph::storage::v3 {
 
 struct Vertex {
-  Vertex(Gid gid, Delta *delta) : gid(gid), deleted(false), delta(delta) {
+  Vertex(Gid gid, Delta *delta) : keys{{PropertyValue{gid.AsInt()}}}, deleted(false), delta(delta) {
     MG_ASSERT(delta == nullptr || delta->action == Delta::Action::DELETE_OBJECT,
               "Vertex must be created with an initial DELETE_OBJECT delta!");
   }
 
-  Gid gid;
+  Gid Gid() const { return Gid::FromInt(keys.GetKey(0).ValueInt()); }
```

Comment:
You don't think its valuable to store gid for quicker retrival?

Reply:
After we have schema, gid will be gone. Currently the "implicit schema" has only one property, the gid.